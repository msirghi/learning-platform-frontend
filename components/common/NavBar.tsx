import React, { useEffect, useState } from 'react';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import Typography from '@material-ui/core/Typography';
import NotificationsIcon from '@material-ui/icons/Notifications';
import styles from '../../styles/modules/Shared.module.scss';
import Divider from '@material-ui/core/Divider';
import { UserAvatar } from './UserAvatar';
import Badge from '@material-ui/core/Badge';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import { LanguageSelect } from './LanguageSelect';
import { useTranslation } from '../../i18n';
import { useSelector } from 'react-redux';
import { RootState } from '../../redux/reducers';

type Props = {
  classes: { [key: string]: string };
  handleDrawerOpen: () => void;
  handleDrawerClose: () => void;
  open: boolean;
};

export const NavBar: React.FC<Props> = ({ classes, handleDrawerOpen, open, handleDrawerClose }) => {
  const [activeTab, setActiveTab] = useState<string | undefined>();
  const { t } = useTranslation();
  const locale = useSelector((state: RootState) => state.preference.locale);

  useEffect(() => {
    /* istanbul ignore next */
    if (global.window) {
      const href = global.window.location.pathname.replaceAll('/', '');
      setTimeout(() => setActiveTab(t(`common:headerTabs.${href}`)), 100);
    }
  }, [locale]);

  return (
    <AppBar
      position='fixed'
      className={clsx(classes.appBar, {
        [classes.appBarShift]: open
      })}
    >
      <Toolbar>
        <IconButton
          color='inherit'
          aria-label='open drawer'
          onClick={handleDrawerOpen}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: open
          })}
        >
          <MenuIcon />
        </IconButton>

        <IconButton
          color='inherit'
          aria-label='close drawer'
          onClick={handleDrawerClose}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: !open
          })}
        >
          <ChevronLeftIcon />
        </IconButton>
        <Typography variant='h6' noWrap>
          {activeTab}
        </Typography>
        <div className={styles.navBarRightIconsContainer}>
          <LanguageSelect />
          <IconButton>
            <Badge badgeContent={4} color='primary'>
              <NotificationsIcon color={'primary'} />
            </Badge>
          </IconButton>
          <IconButton>
            <UserAvatar />
          </IconButton>
        </div>
      </Toolbar>
      <div className={styles.navBarDividerContainer}>
        <Divider light />
      </div>
    </AppBar>
  );
};
