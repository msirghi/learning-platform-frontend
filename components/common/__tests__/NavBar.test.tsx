import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { NavBar } from '../NavBar';
import IconButton from '@material-ui/core/IconButton';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('NavBar component', () => {
  const props = {
    classes: { nav: 'nav' },
    open: true,
    handleDrawerOpen: jest.fn(),
    handleDrawerClose: jest.fn()
  };

  const mockStore = configureStore();
  const store = mockStore({ preference: { locale: 'en' } });

  beforeAll(() => {
    String.prototype.replaceAll = jest.fn();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <NavBar {...props} />
      </Provider>
    );
    expect(toJson(wrapper.find(NavBar))).toMatchSnapshot();
  });

  it('should fire handleDrawerOpen prop on icon button click', () => {
    const wrapper = mount(
      <Provider store={store}>
        <NavBar {...props} />
      </Provider>
    );
    const container = wrapper.find(NavBar);
    const icon = container.find(IconButton).at(0);
    icon.simulate('click');
    expect(props.handleDrawerOpen).toHaveBeenCalled();
  });

  it('should fire handleDrawerClose prop on arrow icon button click', () => {
    const wrapper = mount(
      <Provider store={store}>
        <NavBar {...props} />
      </Provider>
    );
    const container = wrapper.find(NavBar);
    const icon = container.find(IconButton).at(1);
    icon.simulate('click');
    expect(props.handleDrawerClose).toHaveBeenCalled();
  });
});
