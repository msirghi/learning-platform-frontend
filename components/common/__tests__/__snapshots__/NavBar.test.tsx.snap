// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should match the snapshot 1`] = `
<NavBar
  classes={
    Object {
      "nav": "nav",
    }
  }
  handleDrawerClose={[MockFunction]}
  handleDrawerOpen={[MockFunction]}
  open={true}
/>
`;
